### PR TITLE
Add MPI.reduce!

### PR DESCRIPTION
**Pull request by Sbozzolo**
From: https://juliaparallel.org/MPI.jl/stable/reference/collective/#Reduce/Scan

(I still have to test it)

### CODE REVIEW

File: src/mpi.jl [hunk 1]
```diff
@@ -46,6 +46,12 @@ barrier(ctx::MPICommsContext) = MPI.Barrier(ctx.mpicomm)
 
 reduce(ctx::MPICommsContext, val, op) = MPI.Reduce(val, op, 0, ctx.mpicomm)
 
+reduce!(ctx::MPICommsContext, sendbuf, recvbuf, op) =
+    MPI.Reduce!(sendbuf, recvbuf, op, ctx.mpicomm, root = 0)
+
+reduce!(ctx::MPICommsContext, sendrecvbuf, op) =
+    MPI.Reduce!(sendrecvbuf, op, ctx.mpicomm, root = 0)
```
[the remark this answers is not in the document]
Comment (Sbozzolo) [reply src/mpi.jl:53]: Everywhere else in ClimaComms we are assuming that root is 0 and not exposing it (e.g., in `bcast`).

**Review by charleskawczynski — APPROVED** [review]
Want to bump the version for a new release?

**Comment by Sbozzolo** [comment]
> Want to bump the version for a new release?

I added tests, documentation, and bumped the version.

**Comment by Sbozzolo** [comment]
@charleskawczynski want to give a second look?

**Pull request merged**